src/ml_board/frontend/Dashboard/src/worker_socket/MainThreadCallbacks.ts

```diff
@@ -0,0 +1,66 @@
+import { DataFromSocket, DataToRedux } from "./DataTypes";
+import { MLGYM_EVENT } from "./EventsTypes";
+import handleEvaluationResultData, { evalResultCustomData, EvaluationResultPayload } from "./event_handlers/evaluationResultDataHandler";
+import handleExperimentStatusData from "./event_handlers/ExperimentStatusHandler";
+import handleJobStatusData from "./event_handlers/JobStatusHandler";
+
+// Note: I don't like having anything here other than the callbacks but it is what it is :')
+// ========================= variables ============================//
+
+// TODO: temporary until the mystery why this is even used be solved
+const initialStateForGraphs: evalResultCustomData = {
+    grid_search_id: null,
+    experiments: {},
+    colors_mapped_to_exp_id: {}
+};
+
+
+// ========================= Callbacks to update the MainThread ============================//
+// ========================= MTCb == MainThreadCallback ====================================//
+
+export const updateMTCb = (parsedSocketData: DataFromSocket) => {
+    const eventType = parsedSocketData.event_type.toLowerCase();
+    const dataToRedux: DataToRedux = {};
+    switch (eventType) {
```

Comment:
So switch has quite a poor run time performance (same es multiple if statements). 
It's better to have a dictionary mapping `eventType` to the respective "process function", as it has O(1) .

Reply:
Done!

```
// Hashing is faster instead of switching over the the eventType
const MapEventToProcess = {
    [MLGYM_EVENT.JOB_STATUS]: (dataToRedux: DataToRedux, data: any) => { dataToRedux.jobStatusData = handleJobStatusData(data) },
    [MLGYM_EVENT.JOB_SCHEDULED]: () => console.log("Job scheduled found"),
    [MLGYM_EVENT.EVALUATION_RESULT]: (dataToRedux: DataToRedux, data: any) => {
        const evalResPayload = data as unknown as EvaluationResultPayload;
        dataToRedux.evaluationResultsData = handleEvaluationResultData(initialStateForGraphs, evalResPayload);
        dataToRedux.latest_split_metric = evalResPayload;
    },
    [MLGYM_EVENT.EXPERIMENT_CONFIG]: () => console.log("Exp config found"),
    [MLGYM_EVENT.EXPERIMENT_STATUS]: (dataToRedux: DataToRedux, data: any) => { dataToRedux.experimentStatusData = handleExperimentStatusData(data) },
};

// and is used like so
MapEventToProcess[eventType as keyof typeof MapEventToProcess](dataToRedux, parsedSocketData.payload);
```